### ISSUE TITLE
Suggestion: Add BigDecimal.shouldBeCloseTo

### ISSUE THREAD

**Issue by ChristianIvicevic** [MarkusAmshove/Kluent#155]
I ran into a scenario wherein I'd like to assert a `BigDecimal` to be within range of a target value with a specified error margin. Thus I added the following custom extension:

```kotlin
fun BigDecimal.shouldBeCloseTo(
    target: BigDecimal,
    error: BigDecimal
) = shouldBeInRange(target - error, target + error)
```

I expected this to already be a part of the library, but it wasn't. Have you considered adding this to the library?

**Comment by MarkusAmshove** [comment]
That would be a great addition and I will gladly merge a PR if you're willing to contribute  :-)

**Comment by ChristianIvicevic** [comment]
I'll be looking into it, it will take slightly more work for the other numerical types to make it possibly generic if possible, but I'd like to contribute nonetheless once I find the time.

**Comment by MarkusAmshove** [comment]
Thank you for the contribution. I've just released it as v1.55 :-)
